Tcl-tk does not build on 10.9 with gcc@7

This is the error:

Last 15 lines from /Users/carlo/Library/Logs/Homebrew/tcl-tk/06.make:
/private/tmp/tcl-tk--tk-20180610-83460-8gssds/tk8.6.8/unix/../macosx/tkMacOSXXStubs.c:183:2: error: use of undeclared identifier 'NSOperatingSystemVersion'
        NSOperatingSystemVersion systemVersion = [[NSProcessInfo processInfo] operatingSystemVersion];
        ^
/private/tmp/tcl-tk--tk-20180610-83460-8gssds/tk8.6.8/unix/../macosx/tkMacOSXXStubs.c:184:10: error: use of undeclared identifier 'systemVersion'
        major = systemVersion.majorVersion;
                ^
/private/tmp/tcl-tk--tk-20180610-83460-8gssds/tk8.6.8/unix/../macosx/tkMacOSXXStubs.c:185:10: error: use of undeclared identifier 'systemVersion'
        minor = systemVersion.minorVersion;
                ^
/private/tmp/tcl-tk--tk-20180610-83460-8gssds/tk8.6.8/unix/../macosx/tkMacOSXXStubs.c:186:10: error: use of undeclared identifier 'systemVersion'
        patch = systemVersion.patchVersion;
                ^
4 errors generated.
make: *** [tkMacOSXXStubs.o] Error 1
make: *** Waiting for unfinished jobs....

And this is the  gist from brew gist-logs tcl-tk:
https://gist.github.com/e93ff6b8b25ed024c5ef3d3887d16891

Eventually, even using gcc@7, tcl-tk did not build with:

Last 15 lines from /Users/carlo/Library/Logs/Homebrew/tcl-tk/06.make:
                           ^
/private/tmp/tcl-tk--tk-20180610-42357-jjl0ha/tk8.6.8/unix/../macosx/tkMacOSXDialog.c: In function 'Tk_GetSaveFileObjCmd':
/private/tmp/tcl-tk--tk-20180610-42357-jjl0ha/tk8.6.8/unix/../macosx/tkMacOSXDialog.c:996:27: error: expected expression before '^' token
         completionHandler:^(NSInteger returnCode)
                           ^
/private/tmp/tcl-tk--tk-20180610-42357-jjl0ha/tk8.6.8/unix/../macosx/tkMacOSXDialog.c: In function 'Tk_ChooseDirectoryObjCmd':
/private/tmp/tcl-tk--tk-20180610-42357-jjl0ha/tk8.6.8/unix/../macosx/tkMacOSXDialog.c:1146:27: error: expected expression before '^' token
         completionHandler:^(NSInteger returnCode)
                           ^
/private/tmp/tcl-tk--tk-20180610-42357-jjl0ha/tk8.6.8/unix/../macosx/tkMacOSXDialog.c: In function 'Tk_MessageBoxObjCmd':
/private/tmp/tcl-tk--tk-20180610-42357-jjl0ha/tk8.6.8/unix/../macosx/tkMacOSXDialog.c:1429:27: error: expected expression before '^' token
         completionHandler:^(NSModalResponse returnCode)
                           ^
make: *** [tkMacOSXDialog.o] Error 1
make: *** Waiting for unfinished jobs....

This is the new gist from brew gist-logs tcl-tk:
https://gist.github.com/69568c63953482f1f35502cf0c94a4cc

--- a/Formula/tcl-tk.rb
+++ b/Formula/tcl-tk.rb
@@ -13,6 +13,15 @@ class TclTk < Formula
       mirror "ftp://ftp.tcl.tk/pub/tcl/tcl8_6/tk8.6.8-src.tar.gz"
       version "8.6.8"
       sha256 "49e7bca08dde95195a27f594f7c850b088be357a7c7096e44e1158c7a5fd7b33"
+
+      # Upstream issue 7 Jan 2018 "Build failure with Aqua support on OS X 10.8 and 10.9"
+      # See https://core.tcl.tk/tcl/tktview/95a8293a2936e34cc8d0658c21e5214f1ca9b435
+      if MacOS.version == :mavericks || MacOS.version == :mountain_lion
+        patch :p0 do
+          url "https://raw.githubusercontent.com/macports/macports-ports/0a883ad388b/x11/tk/files/patch-macosx-tkMacOSXXStubs.c.diff"
+          sha256 "2cdba6bbf2503307fe4f4d7200ad57c9926ebf0ff6ed3e65bf551067a30a04a9"
+        end
+      end
     end
   end
 

